### PR TITLE
OCPBUGS-18865: add monitortest: in-cluster disruption monitors

### PR DESCRIPTION
**Pull request by vrutkovs**
Re-introduce in-cluster monitors after it was reverted in https://github.com/openshift/origin/pull/28230

### CODE REVIEW

**Comment by vrutkovs** [comment]
/payload-job periodic-ci-openshift-release-master-nightly-4.14-e2e-metal-ipi-ovn-ipv6

**Comment by vrutkovs** [comment]
/payload-job periodic-ci-openshift-release-master-nightly-4.14-e2e-metal-ipi-ovn-ipv6

**Comment by vrutkovs** [comment]
/payload-job periodic-ci-openshift-release-master-nightly-4.14-e2e-metal-ipi-ovn-ipv6

**Comment by vrutkovs** [comment]
/payload-job periodic-ci-openshift-release-master-ci-4.14-upgrade-from-stable-4.13-e2e-aws-ovn-upgrade

**Comment by vrutkovs** [comment]
/payload-job periodic-ci-openshift-release-master-ci-4.14-upgrade-from-stable-4.13-e2e-aws-ovn-upgrade

**Comment by vrutkovs** [comment]
/payload-job periodic-ci-openshift-release-master-ci-4.14-upgrade-from-stable-4.13-e2e-aws-ovn-upgrade

**Comment by vrutkovs** [comment]
/payload-job periodic-ci-openshift-release-master-ci-4.14-upgrade-from-stable-4.13-e2e-aws-ovn-upgrade

**Comment by deads2k** [comment]
what actually panicked and hwo did this fix it?

**Comment by vrutkovs** [comment]
image pullspec was unavailable on metal-ipi jobs (they use mirrored repo). [`bccde11` (#28231)](https://github.com/openshift/origin/pull/28231/commits/bccde11bbe1337b60b9b5b6aa9873fc9770694e8) has fixed it

**Comment by deads2k** [comment]
squash up into one commit unreverting and another commit fixing the problems to make it easier to reivew please.

And looks like you missed setting the notsupportedreason

File: pkg/monitortests/kubeapiserver/incluster_disruption_serializer/remote.go [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+	}
+	i.namespaceName = actualNamespace.Name
+	return nil
+}
+
+func (i *InvariantInClusterDisruption) StartCollection(ctx context.Context, adminRESTConfig *rest.Config, _ monitorapi.RecorderWriter) error {
+	var err error
+	i.openshiftTestsImagePullSpec, i.notSupportedReason, err = i.getImagePullSpec(ctx, adminRESTConfig)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Failed to find test image pullspec: %v\n", err)
```
[the remark this answers is not in the document]
Comment (vrutkovs) [reply pkg/monitortests/kubeapiserver/incluster_disruption_serializer/remote.go:204]: done

**Comment by vrutkovs** [comment]
/payload-job periodic-ci-openshift-release-master-ci-4.14-upgrade-from-stable-4.13-e2e-aws-ovn-upgrade
/payload-job periodic-ci-openshift-release-master-nightly-4.14-e2e-metal-ipi-ovn-ipv6

**Comment by vrutkovs** [comment]
/payload-job periodic-ci-openshift-release-master-ci-4.15-upgrade-from-stable-4.14-e2e-aws-ovn-upgrade periodic-ci-openshift-release-master-nightly-4.14-e2e-metal-ipi-ovn-ipv6

**Comment by vrutkovs** [comment]
/payload-job periodic-ci-openshift-release-master-ci-4.15-upgrade-from-stable-4.14-e2e-aws-ovn-upgrade periodic-ci-openshift-release-master-nightly-4.15-e2e-metal-ipi-ovn-ipv6

**Comment by deads2k** [comment]
some fascinating results on https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/28231/pull-ci-openshift-origin-master-e2e-aws-ovn-single-node-serial/1704540536481779712

/lgtm
/approve
/hold

please check your payload results and if this looks good, feel free to release the hold

**Comment by deads2k** [comment]
panic in this one: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/28231/pull-ci

>caught panic: runtime error: invalid memory address or nil pointer dereference stack trace:
goroutine 418920 [running]:
runtime/debug.Stack()
	runtime/debug/stack.go:24 +0x65
github.com/openshift/origin/pkg/monitortestframework.collectDataWithPanicProtection.func1()
	github.com/openshift/origin/pkg/monitortestframework/panic.go:29 +0x45
panic({0x73c2da0, 0xbae3b10})
	runtime/panic.go:884 +0x213
github.com/openshift/origin/pkg/monitortests/network/disruptionpodnetwork.(*podNetworkAvalibility).CollectData(0xc002736000, {0x8ade1e0, 0xc006a2c5a0}, {0xc006b07d60?, 0xc006b07d60?}, {0x2621e7c?, 0x0?, 0x7da27a0?}, {0xc13af0c7f07816b9, 0x26af1d3aa7d, ...})
	github.com/openshift/origin/pkg/monitortests/network/disruptionpodnetwork/monitortest.go:190 +0x137
github.com/openshift/origin/pkg/monitortestframework.collectDataWithPanicProtection({0x8ade1e0?, 0xc006a2c5a0?}, {0x8ae7920?, 0xc002736000?}, {0x7ffce565e68d?, 0xc00527ded0?}, {0x2647ba5?, 0xc01b5eee40?, 0xcfd71e0?}, {0xc13af0c7f07816b9, ...})
	github.com/openshift/origin/pkg/monitortestframework/panic.go:33 +0x10c
github.com/openshift/origin/pkg/monitortestframework.(*monitorTestRegistry).CollectData.func1({0x8ade1e0, 0xc006a2c5a0}, 0xc002730c90)
	github.com/openshift/origin/pkg/monitortestframework/impl.go:128 +0x248
created by github.com/openshift/origin/pkg/monitortestframework.(*monitorTestRegistry).CollectData
	github.com/openshift/origin/pkg/monitortestframework/impl.go:123 +0x16e


have not yet dug

**Comment by vrutkovs** [comment]
>panic in this one: [prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/28231/pull-ci](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/28231/pull-ci)

this is for networking folks
>github.com/openshift/origin/pkg/monitortests/network/disruptionpodnetwork/monitortest.go:190 +0x137

/hold cancel
/retest-required

**Pull request merged**